### PR TITLE
Enable policy opts via http-api

### PR DESCRIPTION
**Pull request by mpoffald**
Required for https://github.com/fluree/http-api-gateway/issues/38

This PR enables the use of policy opts via http-api. 

The only _required_ change here was https://github.com/fluree/db/commit/56e770de95ca55fb6ef8e334f47897fb01980850.

I also added a change to how we deal with errors when compiling policies (f394d7867dee564b66943c17f3227d79787eb49c). Prior to this change, if there was an error in that codepath, we would fail to enforce any policy at all, silently failing open to root access.  

We _should_  do something smarter with those errors (they are still not surfaced correctly, so I have left the `TODO`s intact), but this was a minimal change that ensures that we at least fail closed when policy compilation fails.

I encountered this earlier in this work, where due to some bugs I was getting errors during the policy compilation process. Since those got fixed I'm no longer getting those errors, and I couldn't readily think of a way to set up a test for this that would demonstrate this. So, we can take it or leave it, since it's untested.

### CODE REVIEW

File: test/fluree/db/policy/parsing_test.clj [hunk 1]
```diff
@@ -111,8 +111,31 @@
                   :roles    #{sid-userRole}}
                  policy-alice)
               "Policies for only :ex/userRole should return")))
-
-
+      (testing "Root policy contains {:root? true} for each applicable :f/action"
+        (let [policy-root  (-> @(fluree/promise-wrap (policy/policy-map db root-did :ex/rootRole nil))
+                               replace-policy-fns)
+              sid-root-did @(fluree/internal-id db root-did)
```
Comment (dpetran) [inline test/fluree/db/policy/parsing_test.clj:117]: Sorry, I'm not very familiar with all the policy work yet, but why do we need to use raw internal subject ids as the :ident?

**Review by dpetran — APPROVED** [review]
This looks good to me. Nice catch on the "internal" query needing to specify the context type in order to not cause problems, I'll keep my eyes open in case we do that elsewhere.

File: test/fluree/db/policy/parsing_test.clj (hunk 1 above)
Comment (mpoffald) [reply test/fluree/db/policy/parsing_test.clj:117]: This test is just checking that the internal implementation of policy works correctly. Users supply `did`s as the `:identity`, and don't need to use internal subject ids at all. Internally, though, we look up the subject id of the "active" identity and use it when calculating which restrictions apply to that identity.

Does that answer the question, or not quite?

Comment (dpetran) [reply test/fluree/db/policy/parsing_test.clj:117]: Ah, yes, now I get it :+1:

**Pull request merged**